packages/dataviews/src/components/dataviews/stories/index.story.tsx

```diff
@@ -184,3 +184,39 @@ export const CombinedFields = () => {
 		/>
 	);
 };
+
+export const PinnedItems = () => {
+	const [ view, setView ] = useState< View >( {
+		...DEFAULT_VIEW,
+		fields: [ 'title', 'description', 'categories' ],
+	} );
+	const { data: shownData, paginationInfo } = useMemo( () => {
+		return filterSortAndPaginate( data, view, fields );
+	}, [ view ] );
+	// Pin the first item
+	const [ pinnedItems, setPinnedItems ] = useState< string[] >( [
+		...shownData[ 0 ].id.toString(),
+	] );
+
+	return (
+		<DataViews
+			getItemId={ ( item ) => item.id.toString() }
+			paginationInfo={ paginationInfo }
+			data={ shownData }
+			enablePinnedItems
+			pinnedItems={ pinnedItems }
```

Comment:
Providing a list of "pinned items" seems to be a good direction for this feature:

- I presume we'd want pinned items to be respected by every layout, like selection is.
- Like selection, it should be a list of ids.